.happy/terraform/modules/sfn/main.tf

```diff
@@ -174,6 +174,12 @@ resource "aws_sfn_state_machine" "state_machine" {
         "Type": "Task",
         "InputPath": "$",
         "Resource": "${var.lambda_error_handler}",
+        "Parameters": {
+          "execution.$": "$$.Execution.Id",
+          "error.$": "$.error",
+          "dataset_id.$": "$.dataset_id",
+          "collection_id.$": "$.collection_id"
+        },
```

Comment:
@Bento007 could you, for ease of review, add some clarity about that `$` and `$$` is doing here? I found [some documentation](https://states-language.net/) but I'm sorry to say it really only makes things like 10% more interpretable to me 😅

Reply:
Fair enough, this took me a while to grep as well. All of the data in a step function is JSON and can be accessed using a query language similar to [JSONPath](https://goessner.net/articles/JsonPath/). I didn't see anywhere that says they officially use this query language, but that's what it looks like.

The `$$` lets you access the [context object](https://docs.aws.amazon.com/step-functions/latest/dg/input-output-contextobject.html) associated with the step function.

The `$` lets you access [input and output values](https://docs.aws.amazon.com/step-functions/latest/dg/input-output-inputpath-params.html) from other steps.

The `.$` suffix is required for the key value if you are getting a value using JSONPath.  From AWS: 
> For key-value pairs where the value is selected using a path, the key name must end in .$.

Reply:
Thanks. This clears things up; I was missing the last bit about the req for the key to have the `.$` suffix.